packages/web/app/src/lib/preflight-sandbox/graphiql-plugin.tsx

```diff
@@ -0,0 +1,550 @@
+import { ComponentProps, Dispatch, useCallback, useEffect, useRef, useState } from 'react';
```

Comment:
@dimaMachina this file seems a bit too messy and heavy. I know `graphiql` plugins are not the easiest, but I'm sure we can break it into smaller, simpler pieces. Can you please iterate it a bit to try to make it simpler and add some comments on the important flows?

Reply:
Also, the state management section seems a bit messy? 🤔